test/lint/lint-assertions.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2018-2022 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#
+# Check for assertions with obvious side effects.
+
+import sys
+import subprocess
+
+
+def exec_cmd(command: [], error_msg: ""):
+    try:
+        output = subprocess.check_output(command, universal_newlines=True, encoding="utf8")
+        print(error_msg)
+        print(output)
+        return 1
+    except subprocess.CalledProcessError as ex1:
+        if ex1.returncode > 1:
```

Comment:
Why check for a return code larger than 1 here? The original script doesn't check return codes at all. As we've discussed in other linter reviews before, it's fine to leave out this `except` and let a `CalledProcessError` escalate. This only hapens if git fails.

Reply:
When the output of grep is empty the return code will be 1, it in the bash script it don't stop the execution because it in a `$()`, that will get only output ignoring the exit code.
The idea here is expect the `CalledProcessError.returncode` equal 1 to have sure if command output is empty.

Reply:
OK, that makes sense. Do you know if this behavior of `git grep` is documented anywhere?